__tests__/core/type-system.test.ts

```diff
@@ -1151,7 +1150,6 @@ test("maybe / optional type inference verification", () => {
   assertTypesEqual(
     _ as ITC,
     _ as {
-      [$nonEmptyObject]?: any
```

Comment:
**praise:** @thegedge - it's very exciting to me to see this gone, this makes the test a lot easier to understand, from my perspective.